### PR TITLE
(GH-354) Updated to patch LibGit2Sharp.dll.config

### LINKED ISSUE #354: Setup task fails on certain linux distros when GitVersion is enabled
On some linux distros GitVersion is unable to run unless a small modification to the `LibGit2Sharp.dll.config` configuration script is patched to allow for a fallback to the system installed libgit2.so file is included (in my case, this happens on Arch Linux, but other distros may also be affected).

I'm working on updating the the build files to patch the necessary file to include than fallback.

### PR DESCRIPTION
**Pull request by AdmiringWorm**
While not exactly perfect (Best would have been to have this included already upstream), this is a workaround (hack) ta allow GitVersion (and Cake.Git) to work on Linux systems that is unable to load the native library included in the nupkg files).

fixes #354 

/CC @gep13 @pascalberger

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
Seems like I need some more testing on other distros before this can be merged

**Comment by AdmiringWorm** [comment]
Seems like the location may differ between distros.
For instance, in Ubuntu Xenial the necessary library is located at `/usr/lib/x86_64-linux-gnu/` and not avaliable without an soname.

Still looking into the best way to support this, and to see if any other major distributions have it located somewhere else.

**Comment by AdmiringWorm** [comment]
I've completed my testing, and believe I've found the necessary paths that libgit2 may appear in. (For now, the Distros tested have been Ubuntu Xenial, OpenSUSE 42 and Arch Linux.)

File: Cake.Recipe/Content/gitversion.cake [hunk 1]
```diff
@@ -24,6 +24,10 @@ public class BuildVersion
 
         if (BuildParameters.ShouldRunGitVersion)
         {
+            if (context.IsRunningOnUnix()) {
```
Comment (gep13) [inline Cake.Recipe/Content/gitversion.cake]: @AdmiringWorm in principle, I have no objection to this getting added, however, I think we might want a way to break out of this happening, if the user doesn't want it.  How about an additional BuildParameter, which is `true` by default added to this if statement.  That way, if can be set to false, if the user doesn't want to do this for some reason.  Thoughts?

Comment (AdmiringWorm) [reply Cake.Recipe/Content/gitversion.cake]: I would thing that if a user wants to be able to run gitversion on linux, they would have no problems with the change.

The time it takes to run the code as well isn't really noticable, and if the the distro is able to load the included libgit library it won't even load the system library at all.

So IMO, I really see no need to have a BuildParameter to enable/disable this.

**Pull request merged**